redbot/cogs/mod/names.py

```diff
@@ -245,15 +245,25 @@ async def userinfo(self, ctx, *, user: discord.Member = None):
         data.add_field(name=_("Joined Discord on"), value=created_on)
         data.add_field(name=_("Joined this server on"), value=joined_on)
         if role_str is not None:
-            data.add_field(name=_("Roles"), value=role_str, inline=False)
+            data.add_field(
+                name=_("Roles") if len(roles) > 2 else _("Role"), value=role_str, inline=False
+            )  # len(roles) > 2 because @everyone is a "role" which is just not displayed.
         if names:
             # May need sanitizing later, but mentions do not ping in embeds currently
             val = filter_invites(", ".join(names))
-            data.add_field(name=_("Previous Names"), value=val, inline=False)
+            data.add_field(
+                name=_("Previous Names") if len(val) > 1 else _("Previous Name"),
```

Comment:
You need to check the length of names list, not the length of the resulting string.
```suggestion
                name=_("Previous Names") if len(names) > 1 else _("Previous Name"),
```